Sources/LottieExtensions/LottieView.swift

```diff
@@ -1,53 +1,132 @@
-        self.filename = filename
-        self.bundle = bundle
-        self.loopMode = loopMode
-        self.completion = completion
-    }
-
-    public func makeUIView(context: UIViewRepresentableContext<LottieView>) -> UIView {
-        let view = UIView()
-        let animation = Animation.named(filename, bundle: bundle)
-        animationView.animation = animation
-        animationView.loopMode = loopMode
-        animationView.contentMode = .scaleAspectFit
-        animationView.backgroundBehavior = .pauseAndRestore
-        if context.environment.disableAnimations {
-            animationView.currentProgress = lottieAnimationProgress
-        } else {
-            animationView.play(completion: completion)
-        }
-        animationView.translatesAutoresizingMaskIntoConstraints = false
-        view.addSubview(animationView)
+    public typealias Configuration = LottieAnimationView
+    private let makeAnimationView: () -> LottieAnimationView
+
+    // Deprecated
+    private let deprecatedCompletion: LottieCompletionBlock?
+    private let runDeprecatedLogic: ((UIView, Context) -> Void)?
+
+    /// Initializes a new `LottieView` with a closure that creates an instance of `LottieAnimationView`.
+    ///
+    /// - Parameter animationView: A closure that creates a `LottieAnimationView` instance.
+    ///
+    /// Example:
+    /// ```swift
+    /// LottieView(Configuration(name: animationFile.name, bundle: animationFile.bundle))
+    /// ```
+    public init(
+        _ animationView: @escaping @autoclosure () -> Configuration
+    ) {
+        self.makeAnimationView = animationView
+        self.deprecatedCompletion = nil
+        self.runDeprecatedLogic = nil
+    }
+
+    public func makeUIView(context: Context) -> UIView {
+        let animationView = makeAnimationView()
+        let wrapperView = UIView()
 
+        animationView.translatesAutoresizingMaskIntoConstraints = false
+        wrapperView.addSubview(animationView)
         NSLayoutConstraint.activate([
-            animationView.heightAnchor.constraint(equalTo: view.heightAnchor),
-            animationView.widthAnchor.constraint(equalTo: view.widthAnchor)
+            animationView.heightAnchor.constraint(equalTo: wrapperView.heightAnchor),
+            animationView.widthAnchor.constraint(equalTo: wrapperView.widthAnchor)
         ])
-        return view
+
+        runDeprecatedLogic?(wrapperView, context)
+        return wrapperView
+    }
+
+    public func updateUIView(_ uiView: LottieView.UIViewType, context: Context) {
+        context.environment.lottieAnimationConfiguration(Self.animationView(uiView))
+        runDeprecatedLogic?(uiView, context)
+    }
+}
+
+// MARK: - Helpers
+extension LottieView {
+    static private func animationView(_ wrapperView: UIView) -> LottieAnimationView {
+        guard let animationView = wrapperView.subviews.first as? LottieAnimationView
+        else {
+            preconditionFailure(
+                """
```

Comment:
Question of taste, but having multiline strings for a single line is a bit ... wasteful? Everything would easily fit in one line.

Reply:
Right, probably forgot while changing string.